File: easy/0492_construct_the_rectangle/solution.cpp
#include <cmath>
#include <vector>

class Solution {
public:
  std::vector<int> constructRectangle(int area) {
    for (int side = std::sqrt(area); side > 0; --side) {
      if (area % side == 0)
        return {area / side, side};
    }

    return {area, 1};
  }
};

File: easy/0492_construct_the_rectangle/solution.py
import math
from typing import List


class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        rectangle = [area, 1]

        for side in range(2, int(math.sqrt(area)) + 1):
            if area % side == 0:
                rectangle = [area // side, side]

        return rectangle
